src/api/ledger/parse/utils.js

```diff
@@ -8,10 +8,11 @@ const BigNumber = require('bignumber.js');
 
 function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
     takerPaysCurrency: string) {
-  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
-              - (takerPaysCurrency === 'XRP' ? 6 : 0);
-  return shift === 0 ? quality :
-    (new BigNumber(quality)).shift(shift).toString();
+  const log10NumeratorAdjustment = (takerPaysCurrency === 'XRP' ? -6 : 0);
```

Comment:
I think the log10*Adjustment names are a bit heavy handed. People know what a shift is, or at least would, with a comment including "e-6 XRP". The brackets around the *adjustment ternary expressions are simply noise. Also, any non zero number is truthy, so you can invert the logic of the return expression:

```javascript
// quality = takerPays/takerGets, using drops (1e-6 XRP) for native operands
const numeratorShift = takerPaysCurrency === 'XRP' ? -6 : 0;
const denominatorShift = takerGetsCurrency === 'XRP' ? -6 : 0;
const shift = numeratorShift - denominatorShift;
return shift ? (new BigNumber(quality)).shift(shift).toString() : 
               quality;
```

As clever as that is, XRP/XRP is ultimately illegal, and asserting the invariants would make more sense than being so accommodating: 

```javascript
assert(!(takerPaysCurrency === 'XRP' &&  takerGetsCurrency === 'XRP'));
```

And having done that it seems like you could jump to the chase, and simply handle the 3 realistic cases:
```javascript
const shift = takerPaysCurrency === 'XRP' ? -6 :  
              takerGetsCurrency === 'XRP' ? 6 : 0;
return shift ? (new BigNumber(quality)).shift(shift).toString() : 
               quality;
```